lib/controllers/cats.js

```diff
@@ -1,3 +1,15 @@
 const { Router } = require('express');
+const { cats } = require('../cats-data');
 
-module.exports = Router();
+module.exports = Router()
+  .get('/:id', (req, res) => {
+    const match = cats.find((cat) => cat.id === req.params.id);
+    return res.json(match);
+  })
+  .get('/', (req, res) => {
+    const filteredData = [];
+    for (const cat of cats) {
+      filteredData.push({ id: cat.id, name: cat.name });
+    }
```

Comment:
Fine to use a for loop, but the .map array method would have also served you well

```js
const filteredData = cats.map((cat) => ({ id: cat.id, name: cat.name }));